### DISCUSSION TITLE
derive vs proxyWithComputed

### DISCUSSION THREAD

**Discussion by targetlucked69** [pmndrs/valtio#601] [Q&A]
Hello, thanks for this awesome library!

I just want to ask if there's a big difference between [derive](https://github.com/pmndrs/valtio#derive-util) and [proxyWithComputed](https://github.com/pmndrs/valtio#proxywithcomputed-util) in terms of using it with Vanilla valtio?

Thanks!

**Comment by dai-shi** [comment]
There are some subtle differences.
My suggestion is to use `derive` first, and only reach out `proxyWithComputed` if something doesn't work as expected.

**Reply by targetlucked69** [reply]
Thanks @dai-shi! Can you give 1 example of a subtle difference? It's really hard to think of one

**Reply by dai-shi** [reply]
`derive` can create a new proxy, whereas `proxyWithCompute` can only attach properties to the self proxy.
